packages/state/src/index.ts

```diff
@@ -1760,21 +1811,11 @@ export const groupMachine = createMachine(
         items: ({ context, event }) => {
           isEvent(event, ["setActualItemsSize"]);
 
-          const orientation = context.orientation;
-
-          for (const [id, size] of Object.entries(event.childrenSizes)) {
-            const item = context.items.find((i) => i.id === id);
-
-            if (!isPanelData(item)) {
-              continue;
-            }
-
-            item.currentValue = makePixelUnit(
-              orientation === "horizontal" ? size.width : size.height
-            );
```

Comment:
The first big change in this PR is that when we get the actual size of the panel we don't immediately set its size to that. We only use the actual size during resizing operations.

If we update immediately the percentages get updates as the container resizes. Which leads to a bunch of undesired behaviors:

- When a panel renders at 0 initially, it can never recover
- when a panel is resized to 0 it can never recover
- if you resize it updates the percentages and your panels end up in different places

So instead we don't updates the percentages at all and instead just track what size it was last rendered at.